### PR TITLE
port describe-vector

### PR DESCRIPTION
**Pull request by brotzeit**

### CODE REVIEW

File: rust_src/src/keymap.rs [hunk 1]
```diff
@@ -565,4 +568,42 @@ pub fn make_sparse_keymap(string: LispObject) -> LispObject {
...
+    unsafe { Findent_to(LispObject::from_fixnum(16), LispObject::from_fixnum(1)) };
+    call!(fun, elt);
+    unsafe { Fterpri(Qnil, Qnil) };
+}
+
+/// Insert a description of contents of VECTOR.
+/// This is text showing the elements of vector matched against indices.
+/// DESCRIBER is the output function used; nil means use `princ'.
+#[lisp_fn(min = "1", name = "describe-vector")]
+pub fn describe_vector_lisp(vector: LispObject, mut describer: LispObject) {
```
Comment (shaleh) [inline rust_src/src/keymap.rs]: I recently introduced `map_or` and `map_or_else` on `LispObject`. However, neither really fits this situation. I am not sold on `unwrap_or_else` as the right name. Something like `obj.or_default(|| action())` maybe?

**Comment by shaleh** [comment]
@brotzeit I updated this for the 2018 changes.

File: rust_src/src/keymap.rs [hunk 2]
```diff
@@ -569,4 +575,42 @@ pub fn make_sparse_keymap(string: LispObject) -> LispObject {
...
```
Comment (nitishch) [inline rust_src/src/keymap.rs]: Why not name the function `describe_vector`?

Comment (nitishch) [reply rust_src/src/keymap.rs]: I see. There is a C function named `describe_vector`. Sorry for the confusion.

**Comment by brotzeit** [comment]
I don't get it. Why is `void (*elt_describer) (Lisp_Object, Lisp_Object),` an option in rust.

**Comment by shaleh** [comment]
My understanding is bindgen represents pointers to functions this way. None is NULL.

https://github.com/rust-lang-nursery/rust-bindgen/issues/1278

**Pull request merged**